workspace/exercise_metadata.go

```diff
@@ -30,41 +30,41 @@ type Solution struct {
 	AutoApprove bool       `json:"auto_approve"`
 }
 
-// NewSolution reads solution metadata from a file in the given directory.
-func NewSolution(dir string) (*Solution, error) {
+// NewExerciseMetadata reads exercise metadata from a file in the given directory.
+func NewExerciseMetadata(dir string) (*ExerciseMetadata, error) {
 	b, err := ioutil.ReadFile(filepath.Join(dir, metadataFilepath))
 	if err != nil {
-		return &Solution{}, err
+		return &ExerciseMetadata{}, err
```

Comment:
Seeing as our return value is a pointer we can `return nil, err` not bother allocating memory for an empty ExerciseMetadata.